radix-engine-tests/tests/royalty_edge_cases.rs

```diff
@@ -0,0 +1,448 @@
+mod package_loader;
+
+use package_loader::PackageLoader;
+use radix_engine::errors::*;
+use radix_engine::transaction::*;
+use radix_engine_queries::typed_substate_layout::*;
+use scrypto_unit::*;
+use transaction::prelude::*;
+
+#[test]
+fn test_component_royalty_edge_cases_at_instantiation() {
+    // Arrange
+    let mut test_runner = TestRunnerBuilder::new().without_trace().build();
+    let package_address =
+        test_runner.publish_package_simple(PackageLoader::get("royalty-edge-cases"));
+
+    for (royalty_amount, error_checking_fn, _) in test_cases().into_iter() {
```

Comment:
I think cleaner to have these tests cases in separate tests. Also to enforce the property of "isolation" in unit tests. But overall, good enough for now :)

Reply:
Agreed, will refactor into separate tests before merging. I honestly wish Rust had parameterized tests like most other languages!